doc/sphinxext/gallery_generator.py

```diff
@@ -200,6 +235,12 @@ def __init__(self, filename, target_dir, backend, thumb_dir):
 
             print("skipping {0}".format(self.filename))
 
+    @property
+    def title(self) -> str:
+        if self._title is not None:
```

Comment:
`Optional[str]` implies a default of `None`? Or do you have to `_title: Optional[str] = None`?

Reply:
If I do that, I think I end up with a class property instead of an instance property, don't I?

`Optional[str]` does not imply anything about the value.  I will set it to `None` in the `__init__`.